comp/core/workloadmeta/collectors/internal/process/process_collector_test.go

```diff
@@ -0,0 +1,280 @@
+		proc, err := c.mockStore.GetProcess(2)
+		assert.NoError(cT, err)
+		assert.Equal(cT, expectedProc2, proc)
+
+		_, err = c.mockStore.GetProcess(1)
+		assert.Error(cT, err)
+	}, time.Second, time.Millisecond*100)
+}
+
+func TestProcessCollectorStart(t *testing.T) {
+	tests := []struct {
+		name                 string
+		agentFlavor          string
+		langDetectionEnabled bool
+		runInCoreAgent       bool
+		expectedEnabled      bool
+	}{
+		{
+			name:                 "core agent + all configs enabled",
+			agentFlavor:          flavor.DefaultAgent,
+			langDetectionEnabled: true,
+			runInCoreAgent:       true,
+			expectedEnabled:      true,
+		},
+		{
+			name:                 "core agent + all configs disabled",
+			agentFlavor:          flavor.DefaultAgent,
+			langDetectionEnabled: false,
+			runInCoreAgent:       false,
+			expectedEnabled:      false,
+		},
+		{
+			name:                 "process agent + all configs enabled",
+			agentFlavor:          flavor.ProcessAgent,
+			langDetectionEnabled: true,
+			runInCoreAgent:       true,
+			expectedEnabled:      false,
+		},
+		{
+			name:                 "process agent + all configs disabled",
+			agentFlavor:          flavor.ProcessAgent,
+			langDetectionEnabled: false,
+			runInCoreAgent:       false,
+			expectedEnabled:      false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			originalFlavor := flavor.GetFlavor()
+			defer flavor.SetFlavor(originalFlavor)
+			flavor.SetFlavor(test.agentFlavor)
+
+			configOverrides := map[string]interface{}{
+				"language_detection.enabled":               test.langDetectionEnabled,
+				"process_config.run_in_core_agent.enabled": test.runInCoreAgent,
+			}
+
+			c := setUpCollectorTest(t, configOverrides)
+			ctx, cancel := context.WithCancel(context.TODO())
+			defer cancel()
+
+			err := c.collector.Start(ctx, c.mockStore)
+
+			enabled := err == nil
+			assert.Equal(t, test.expectedEnabled, enabled)
+		})
+	}
+}
+
+func TestProcessCollectorWithoutProcessCheck(t *testing.T) {
+	originalFlavor := flavor.GetFlavor()
+	defer flavor.SetFlavor(originalFlavor)
+	flavor.SetFlavor(flavor.DefaultAgent)
+
+	configOverrides := map[string]interface{}{
+		"language_detection.enabled":                true,
+		"process_config.process_collection.enabled": false,
+		"process_config.run_in_core_agent.enabled":  true,
+	}
+
+	c := setUpCollectorTest(t, configOverrides)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	mockCtrl := gomock.NewController(t)
+	mockProvider := proccontainers.NewMockContainerProvider(mockCtrl)
+	c.collector.containerProvider = mockProvider
+
+	err := c.collector.Start(ctx, c.mockStore)
+	require.NoError(t, err)
+
+	c.probe.On("ProcessesByPID", mock.Anything, mock.Anything).Return(map[int32]*procutil.Process{
+		1: {
+			Pid:     1,
+			Cmdline: []string{"proc", "-h", "-v"},
+			Stats:   &procutil.Stats{CreateTime: 1},
+		},
+	}, nil).Maybe()
```

Comment:
Maybe? Shouldn't this definitely get called?

Reply:
fixed